packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -386,3 +401,536 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
+  ///          isAlwaysShown: true,
+  ///          controller: _controllerTwo,
+  ///          child: SingleChildScrollView(
+  ///            controller: _controllerTwo,
+  ///            child: SizedBox(height: 2000, width: 500,),
+  ///          ),
+  ///        ),
+  ///      ),
+  ///    ],
+  ///   );
+  /// }
+  /// ```
+  /// {@end-tool}
+  final bool isAlwaysShown;
+
+  /// [Radius] of corners if the scrollbar should have rounded corners.
+  ///
+  /// Scrollbar will be rectangular if [radius] is null, which is the default
+  /// behavior.
+  final Radius? radius;
+
+  /// The thickness of the scrollbar in the cross axis of the scrollable.
+  ///
+  /// Cannot be null.
+  final double thickness;
+
+  /// The color of the scrollbar thumb.
+  ///
+  /// Cannot be null.
+  final Color thumbColor;
+
+  /// The [Duration] of the fade animation.
+  ///
+  /// Cannot be null, defaults to a [Duration] of 300 milliseconds.
+  final Duration fadeDuration;
+
+  /// The [Duration] of time until the fade animation begins.
+  ///
+  /// Cannot be null, defaults to a [Duration] of 600 milliseconds.
+  final Duration timeToFade;
+
+  /// The [Duration] of time that a LongPress will trigger the drag gesture of
+  /// the scrollbar thumb.
+  ///
+  /// Cannot be null, defaults to [Duration.zero].
+  final Duration pressDuration;
+
+  @override
+  RawScrollbarState<RawScrollbar> createState() => RawScrollbarState<RawScrollbar>();
+}
+
+/// The state for a [RawScrollbar] widget, also shared by the [Scrollbar] and
+/// [CupertinoScrollbar] widgets.
+///
+/// Controls the animation that fades a scrollbar thumb in and out of view.
+class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProviderStateMixin<T> {
+  double? _dragScrollbarAxisPosition;
+  Drag? _drag;
+  ScrollController? _currentController;
+
+  /// A [GlobalKey] associated with the [painter], will be assigned to the
+  /// [CustomPaint] of the scrollbar.
+  @protected
+  GlobalKey get customPaintKey => _customPaintKey;
+  final GlobalKey _customPaintKey = GlobalKey();
+
+
+  /// Used to paint the scrollbar.
+  @protected
+  ScrollbarPainter? painter;
+
+  /// Controls the fading in and out of the scrollbar thumb.
+  @protected
+  AnimationController get fadeoutAnimationController => _fadeoutAnimationController;
+  late AnimationController _fadeoutAnimationController;
+  
+  /// Used to animate the opacity of the scrollbar thumb.
+  @protected
+  Animation<double> get fadeoutOpacityAnimation => _fadeoutOpacityAnimation;
+  late Animation<double> _fadeoutOpacityAnimation;
+
+  /// Used to initiate the fade out animation upon completion.
+  @protected
+  Timer? get fadeoutTimer => _fadeoutTimer;
+  Timer? _fadeoutTimer;
+  
+  @override
+  void initState() {
+    super.initState();
+    _fadeoutAnimationController = AnimationController(
+      vsync: this,
+      duration: widget.fadeDuration,
+    );
+    _fadeoutOpacityAnimation = CurvedAnimation(
```

Comment:
This is the sort of thing subclasses might want to override. Maybe just have `Animation<double> get fadeOutOpacityAnimation()`, and allow developers to override it to return their own (for which they'd have to manage the lifetime of).